presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -181,6 +187,7 @@ private boolean exploreNode(int group, Context context, Matcher matcher)
                         }
                     }
                     node = context.memo.replace(group, transformedNode, rule.getClass().getName());
+                    rulesTriggered.add(rule.getClass().getSimpleName());
```

Comment:
can we add to the optimizerInformationCollector directly here?

Reply:
This line will be executed for every node which has the optimizer rule triggered, i.e. if we have 10 nodes triggering a rule, it will be executed 10 times. That's why I am using a set to collect the rules here.